platform/config/go-links.yaml

```diff
@@ -22,3 +22,11 @@
 /tools: /documentation/tools
 /vision-mission: /about/mission-and-vision
 /websites: /about/websites
+
+# Regex-based entries
+/c/amp-([a-z-]+):
+  url: /documentation/components/amp-$1
+  useRegex: true
```

Comment:
Seems to complicated to me.
Why don't we simply put the matches in two categories like the code uses them:
```yaml
simple-matches:
   /ads: /about/ads
regex-matches:
  /c/amp-([a-z-]+): /documentation/components/amp-$1
```

We could also simply use regex for all the matches.
The existing go links would even work without any escaping since they do not have special characters like the dot.
The downside is of course, that its easy to forget special escaping might be needed, since it is not obvious that the matches are regex when most of them do not contain regex specific matching.
So I would prefer the categories, but I am open for discussions.

Reply:
Some prior discussion in #2182 - I prefer the object notation since this allows us to add additional features / flags in the future.

I don't see it as more / less complicated than splitting into `simple-matches` and `regex-matches`, since users can easily be confused by what "simple" means. More importantly, I feel there's very limited need for regex matches, so I'm not too concerned about it being slightly more verbose.

> We could also simply use regex for all the matches.

I'm strongly against this for several reasons:

-   It should remain easy for people with no technical background to edit this file and add new links. Regex raises that bar significantly.

-   It's very error-prone: in fact, I just realized that the regex I used is incorrect (lacks `$` at the end).

-   What happens if two regexes match? This can easily happen by accident and it's very difficult to test.

-   Once again, I feel the need for regex matches in redirects is limited, definitely not something most redirects would need.

Reply:
Ok, you convinced me.
Additional flags is a strong point for your config structure.
And also the very obvious difference between direct matches and the few special regexp cases is not so bad.